java/client/src/main/java/glide/api/commands/ScriptingAndFunctionsBaseCommands.java

```diff
@@ -128,4 +132,152 @@ CompletableFuture<Object> fcall(
+     * EVALSHA</code> command.
+     *
+     * @apiNote When in cluster mode
+     *     <ul>
+     *       <li>all <code>keys</code> must map to the same hash slot.
+     *       <li>if no <code>keys</code> are given, command will be routed to a random primary node.
+     *     </ul>
+     *
+     * @see <a href="https://valkey.io/commands/script-load/">SCRIPT LOAD</a> and <a
+     *     href="https://valkey.io/commands/evalsha/">EVALSHA</a> for details.
+     * @param script The Lua script to execute.
+     * @param options The script option that contains keys and arguments for the script.
+     * @return a value that depends on the script that was executed.
+     * @example
+     *     <pre>{@code
+     * try(Script luaScript = new Script("return { KEYS[1], ARGV[1] }", false)) {
+     *     ScriptOptions scriptOptions = ScriptOptions.builder().key("foo").arg("bar").build();
+     *     Object[] result = (Object[]) client.invokeScript(luaScript, scriptOptions).get();
+     *     assert result[0].equals("foo");
+     *     assert result[1].equals("bar");
+     * }
+     * }</pre>
+     */
+    CompletableFuture<Object> invokeScript(Script script, ScriptOptions options);
+
+    /**
+     * Invokes a Lua script with its keys and arguments.<br>
+     * This method simplifies the process of invoking scripts on the server by using an object that
+     * represents a Lua script. The script loading, argument preparation, and execution will all be
+     * handled internally. If the script has not already been loaded, it will be loaded automatically
+     * using the <code>SCRIPT LOAD</code> command. After that, it will be invoked using the <code>
+     * EVALSHA</code> command.
+     *
+     * @apiNote When in cluster mode
+     *     <ul>
+     *       <li>all <code>keys</code> must map to the same hash slot.
+     *       <li>if no <code>keys</code> are given, command will be routed to a random primary node.
+     *     </ul>
+     *
+     * @see <a href="https://valkey.io/commands/script-load/">SCRIPT LOAD</a> and <a
+     *     href="https://valkey.io/commands/evalsha/">EVALSHA</a> for details.
+     * @param script The Lua script to execute.
+     * @param options The script option that contains keys and arguments for the script.
+     * @return a value that depends on the script that was executed.
+     * @example
+     *     <pre>{@code
+     * try(Script luaScript = new Script(gs("return { KEYS[1], ARGV[1] }", true))) {
+     *     ScriptOptionsGlideString scriptOptions = ScriptOptionsGlideString.builder().key(gs("foo")).arg(gs("bar")).build();
+     *     Object[] result = (Object[]) client.invokeScript(luaScript, scriptOptions).get();
+     *     assert result[0].equals(gs("foo"));
+     *     assert result[1].equals(gs("bar"));
+     * }
+     * }</pre>
+     */
+    CompletableFuture<Object> invokeScript(Script script, ScriptOptionsGlideString options);
```

Comment:
Shouldn't be called `ScriptOptionsBinary` or `ScriptBinaryOptions` to align with others?

Reply:
As another option is to get rid of options at all and to such variants of `invoke` API:
- script, keys, args
- script, args, route (for cluster client)
- same 2 with glide strings

In case if user don't need to give keys to the script, the API call will be like `client.invokeScript(script, new String[0], args)`

This will make this API more aligned across standalone and cluster clients